src/v/cloud_storage/segment_meta_cstore.cc

```diff
@@ -230,6 +296,129 @@ class column_store
         }
     }
 
+    auto serde_fields() { return std::tuple_cat(columns(), std::tie(_hints)); }
```

Comment:
1. does this now create a copy?
2. it seems error prone: with serde_fields we basically only get away with appends. but with this formulation it would seem we could only add hints without needing to abandon serde fields mechanism?

Reply:
>could you elaborate on this point?

what is the intended way to extend the data structure? in this formulation you could only append to `_hints` with a basic version bump

Reply:
if i understand the question correctly: yes like in  other cases, new fields would have to be added ` ... std::tie(_hints, _new_field)`
But probably if there is a need to change column_store, chances are that it will be done in serde_write/read. 

...So probably it's another argument to ditch serde_fields here and use another name